backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/BridgesBuilding.kt

```diff
@@ -159,7 +158,7 @@ internal class BridgesBuilding(val context: Context) : ClassLoweringPass {
                 val irBuilder = context.createIrBuilder(declaration.symbol, declaration.startOffset, declaration.endOffset)
                 declaration.body = irBuilder.irBlockBody(declaration) {
                     buildTypeSafeBarrier(declaration, declaration, typeSafeBarrierDescription)
-                    body.statements.forEach { +it }
+                    (body as IrBlockBody).statements.forEach { +it }
```

Comment:
This lowering is before returns insertion, so the former shouldn't be affected by the latter. What is this patch for then?

Reply:
Because earlier we assumed that `IrExpressionBody` is not present in IR anywhere. This is not true anymore. Here we're lowering that condition to only the functions for which `typeSafeBarrierDescription` is not null.

Reply:
Do you mean that if `typeSafeBarrierDescription` is not null, then the body is not `IrExpressionBody`?

Reply:
Yes.